cpp/src/arrow/engine/substrait/serde_test.cc

```diff
@@ -402,22 +402,36 @@ TEST(Substrait, SupportedLiterals) {
   auto ExpectEq = [](std::string_view json, Datum expected_value) {
     ARROW_SCOPED_TRACE(json);
 
-    ASSERT_OK_AND_ASSIGN(
-        auto buf, internal::SubstraitFromJSON("Expression",
-                                              "{\"literal\":" + std::string(json) + "}"));
-    ExtensionSet ext_set;
-    ASSERT_OK_AND_ASSIGN(auto expr, DeserializeExpression(*buf, ext_set));
+    for (bool nullable : {false, true}) {
+      std::string json_with_nullable;
+      std::string nullable_json = (nullable) ? ", \"nullable\": true" : "";
+      if (nullable) {
+        auto final_closing_brace = json.find_last_of('}');
+        ASSERT_NE(std::string_view::npos, final_closing_brace);
+        json_with_nullable =
+            std::string(json.substr(0, final_closing_brace)) + ", \"nullable\": true}";
+        json = json_with_nullable;
+      }
```

Comment:
IIUC, this results in
```json
{"literal": {"boolean": true, "nullable": true}, "nullable": true}
```
but literal_type a `oneof` so it should only have a single key-value pair, where the key names the member of the union:
```json
{"literal": {"boolean": true}, "nullable": true}
```
If this isn't being caught in tests maybe there's a flag "error on unused fields" we should set.
```suggestion
```

Reply:
I see ignoring unknown fields was added in #13605 since the json conversion is being used in more places. This should probably be an argument to `SubstraitFromJSON` instead, so that we can be stricter while testing.

Reply:
Whoops, yeah, `nullable_json` was an earlier attempt and should just be removed.  The point on ignoring unknown fields is a good one though.  I'll make a crack at it.

Reply:
I've added the bool argument to `SubstraitFromJSON` and set it to false for all unit test calls.  To my pleasant surprise, my addition was the only new failure.  I removed `nullable_json`.